Realm/RLMSchema.mm

```diff
@@ -126,12 +126,20 @@ - (instancetype)init {
 
 - (NSArray *)objectSchema {
     if (!_objectSchema) {
+        if (self == s_sharedSchema) {
+            // Ensure the schema is fully initialized
+            [RLMSchema sharedSchema];
```

Comment:
We're not pretending it's the full object schema. The entire point of `partialSharedSchema` is that it gives you the schema loaded so far and not the full one.